site/docs/components/link.mdx

```diff
@@ -96,25 +97,166 @@ Standalone links are used on their own or directly after the content, and they a
+      size="M" 
+      external 
+      openInNewTab
+      openInExternalDomainAriaLabel="Opens a different website."
+      openInNewTabAriaLabel="Opens in a new tab."
+      style={{ display: 'block', width: 'fit-content' }}>
+      Link that opens in a new tab
+    </Link>
+</Playground>
+
+#### Core:
+
+```html
+<a href="https://hds.hel.fi/" class="hds-link hds-link--medium" style="display:block;">Internal link</a>
+
+<a href="https://github.com/City-of-Helsinki/helsinki-design-system" class="hds-link hds-link--medium" aria-label="Opens a different website.">
+  External link <i class="hds-icon icon hds-icon--link-external hds-icon--size-s vertical-align-small-or-medium-icon" aria-hidden="true"></i>
+</a>
+
+<a href="https://github.com/City-of-Helsinki/helsinki-design-system" class="hds-link hds-link--medium" rel="noopener" target="_blank" aria-label="Opens a different website. Opens in a new tab.">
+  Link that opens in a new tab <i class="hds-icon icon hds-icon--link-external hds-icon--size-xs vertical-align-small-or-medium-icon" aria-hidden="true"></i>
+</a>
+```
+
+#### React:
+
+```tsx
+<Link 
+  href="https://hds.hel.fi/" 
+  size="M" 
+  style={{ display: 'block', width: 'fit-content' }}
+>
+  Internal link
+</Link>
+
+<Link 
+  href="https://github.com/City-of-Helsinki/helsinki-design-system" 
+  size="M" 
+  external
+  openInExternalDomainAriaLabel="Opens a different website."
+  style={{ display: 'block', width: 'fit-content' }}
+>
+  External link
+</Link>
+
+<Link 
+  href="https://github.com/City-of-Helsinki/helsinki-design-system" 
+  size="M" 
+  external 
+  openInNewTab
+  openInExternalDomainAriaLabel="Opens a different website."
+  openInNewTabAriaLabel="Opens in a new tab."
+  style={{ display: 'block', width: 'fit-content' }}>
+  Link that opens in a new tab
+</Link>
+```
+
+### Links with icons
+HDS Links can be paired with an icon. This can be useful if the link has unusual target such as a phone number, an email address (mailto link), a PDF document or a photo.
+
+Note! These icons are meant to describe the link target. If you need a directional icon for the link (e.g. an arrow), you should instead use [a supplementary button](/components/button#supplementary-button) with `role="link"`.
+
+<Playground>
+    <Link iconLeft={<IconDocument size="s" aria-hidden />} size="M" href="/#" style={{ display: 'block', width: 'fit-content' }}>
+      Document link
+    </Link>
+    <Link iconLeft={<IconPhone size="s" aria-hidden />} size="M" href="/#" style={{ display: 'block', width: 'fit-content' }}>
+      Phone link
+    </Link>
+    <Link iconLeft={<IconEnvelope size="s" aria-hidden />} size="M" href="/#" style={{ display: 'block', width: 'fit-content' }}>
+      Email link
+    </Link>
+    <Link iconLeft={<IconPhoto size="s" aria-hidden />} size="M" href="/#" style={{ display: 'block', width: 'fit-content' }}>
+      Photo link
+    </Link>
+</Playground>
+
+#### Core:
```

Comment:
@VilleMiekkoja Are these Core code examples ok? These Icon Links were missing from the Core Storybook so I'm wondering are they supported or not?